src/classes/BDI_MatchDonations.cls

```diff
@@ -259,8 +259,8 @@ public with sharing class BDI_MatchDonations implements BDI_IMatchDonations {
                             }
                         }
                     }
-                } else {
-                    di.DonationImportStatus__c = label.bdiMatchedNone;
+                } else if (di.DonationImportStatus__c != Label.bdiMatchedByUser) {
```

Comment:
@bethbrains - what is this doing? Is it just blocking against setting the status if the user has explicitly chosen the match? Do we still need this with the button-level blocking?